packages/query/src/index.ts

```diff
@@ -1316,41 +1318,63 @@ const generateQueryHook = async (
       );
     };
 
-    const queryKeyProps = makeParamsOptional(
-      toObjectString(
-        props.filter((prop) => prop.type !== GetterPropType.HEADER),
-        'implementation',
-      ),
+    const uniqueQueryOptionsByKeys = queries.filter(
+      (obj, index, self) =>
+        index ===
+        self.findIndex((t) => t.queryKeyFnName === obj.queryKeyFnName),
     );
 
-    const routeString =
-      isVue(outputClient) || override.query.shouldSplitQueryKey
-        ? getRouteAsArray(route) // Note: this is required for reactivity to work, we will lose it if route params are converted into string, only as array they will be tracked // TODO: add tests for this
-        : `\`${route}\``;
-
-    // Use operation ID as query key if enabled, otherwise use route string
-    const queryKeyIdentifier = override.query.useOperationIdAsQueryKey
-      ? operationName
-      : routeString;
-
-    // Note: do not unref() params in Vue - this will make key lose reactivity
-    const queryKeyFn = `${
-      override.query.shouldExportQueryKey ? 'export ' : ''
-    }const ${queryKeyFnName} = (${queryKeyProps}) => {
-    return [${queryKeyIdentifier}${queryParams ? ', ...(params ? [params]: [])' : ''}${
-      body.implementation ? `, ${body.implementation}` : ''
-    }] as const;
-    }`;
-
-    implementation += `${queryKeyMutator ? '' : queryKeyFn}
-
-    ${queries.reduce(
-      (acc, queryOption) =>
+    implementation += `
+${
+  !queryKeyMutator
+    ? uniqueQueryOptionsByKeys.reduce((acc, queryOption) => {
+        const queryKeyProps = makeParamsOptional(
+          toObjectString(
+            props.filter((prop) => prop.type !== GetterPropType.HEADER),
+            'implementation',
+          ),
+        );
+
+        const routeString =
+          isVue(outputClient) || override.query.shouldSplitQueryKey
+            ? getRouteAsArray(route) // Note: this is required for reactivity to work, we will lose it if route params are converted into string, only as array they will be tracked // TODO: add tests for this
+            : `\`${route}\``;
+
+        // Use operation ID as query key if enabled, otherwise use route string
+        const queryKeyIdentifier = override.query.useOperationIdAsQueryKey
+          ? operationName
+          : routeString;
+
+        // Note: do not unref() params in Vue - this will make key lose reactivity
+        const queryKeyFn = `
+${override.query.shouldExportQueryKey ? 'export ' : ''}const ${queryOption.queryKeyFnName} = (${queryKeyProps}) => {
+    return [
+    ${[
+      queryOption.type === QueryType.INFINITE ||
+      queryOption.type === QueryType.SUSPENSE_INFINITE
+        ? `'infinate'`
+        : '',
+      queryKeyIdentifier,
+      queryParams ? '...(params ? [params]: [])' : '',
+      body.implementation,
```

Comment:
Previously, the value was used only if `body.implementation` existed, as shown below. Is it okay to remove this existence check?

`body.implementation ? `, ${body.implementation}` : ''`